src/Common/Core/Impl/Services/IServiceManager.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Common.Core.Services {
+    public interface IServiceManager: IServiceContainer, IDisposable {
+        /// <summary>
+        /// Adds service instance
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="service">Service instance</param>
+        IServiceContainer AddService<T>(T service) where T : class;
+
+        /// <summary>
+        /// Adds on-demand created service
+        /// </summary>
+        /// <param name="type">Service type</param>
+        /// <param name="factory">Optional service factory. If not provided, reflection with default constructor will be used.</param>
+        /// <param name="factoryParameters">Optional arguments to the factory call.</param>
+        IServiceContainer AddService(Type type, IServiceFactory factory = null, params object[] factoryParameters);
```

Comment:
I.e. I will be using this instead of static service manager. So `ServiceManager.GetService<T>(buffer)` will be replaced by `buffer.Services.GetService<T>()`. ditto for the view.